code/controllers/subsystem/gravity.dm

```diff
@@ -0,0 +1,180 @@
+
+/datum/subsystem/gravity/Recover()
+	Initialize()	//Force init..
+	if(istype(SSgravity))
+		do_purge = SSgravity.do_purge
+		purge_interval = SSgravity.purge_interval
+		error_no_atom = SSgravity.error_no_atom
+		error_mismatched_area = SSgravity.error_mismatched_area
+		error_mismatched_turf = SSgravity.error_mismatched_turf
+		error_no_area = SSgravity.error_no_area
+		error_no_turf = SSgravity.error_no_turf
+	..()
+
+/datum/subsystem/gravity/proc/sync_to_global_variables()
+	world << "Syncing to global vars!"
+	mob_base_gravity_slip_chance = mob_slip_chance
+	mob_base_gravity_fall_chance = mob_fall_chance
+	mob_handhold_gravity_slip_chance = mob_slip_chance_handhold
+	mob_handhold_gravity_fall_chance = mob_fall_chance_handhold
+	mob_gravity_strength_slip_mod = mob_slip_chance_mod
+	mob_gravity_strength_fall_mod = mob_fall_chance_mod
+	legacy_gravity = normal_gravity
+	if(legacy_gravity)
+		SSgravity.can_fire = FALSE
+	else
+		SSgravity.can_fire = TRUE
+
+/datum/subsystem/gravity/vv_edit_var()
+	..()
+	sync_to_global_variables()
+
+/datum/subsystem/gravity/SDQL_update()
+	..()
+	sync_to_global_variables()
+
+/proc/emergency_reset_gravity_force_processing()
+	var/count = 0
+	while(atoms_forced_gravity_processing.len)
+		var/atom/movable/AM = atoms_forced_gravity_processing[atoms_forced_gravity_processing.len]
+		atoms_forced_gravity_processing.len--
+		if(!istype(AM))
+			if(SSgravity)
+				SSgravity.error_no_atom++
+			continue
+		else
+			AM.force_gravity_processing = FALSE
+			count++
+		CHECK_TICK
+	LAZYCLEARLIST(atoms_forced_gravity_processing)
+	var/atoms_not_found = "ERROR: NO SUBSYSTEM!"
+	if(SSgravity)
+		atoms_not_found = SSgravity.error_no_atom
+	return "[count] atoms purged from forced processing! [atoms_not_found] things found so far that were not atoms!"
+
+/datum/subsystem/gravity/proc/recalculate_atoms()
+	currentrun = list()
+	currentrun_manual = list()
+	var/tempcost = REALTIMEOFDAY
+	for(var/I in sortedAreas)
+		var/area/A = I
+		if(!A.has_gravity && !A.gravity_generator && !A.gravity_overriding)
+			continue
+		if(!A.gravity_direction)	//Right now we don't need this.
+			continue
+		if(is_type_in_typecache(A, area_blacklist_typecache))
+			continue
+		currentrun += A.contents_affected_by_gravity
+	currentrun_manual = atoms_forced_gravity_processing.Copy()
+	recalculation_cost = REALTIMEOFDAY - tempcost
+
+/datum/subsystem/gravity/fire(resumed = FALSE)
+	if(!resumed)
+		if(legacy_gravity)
+			can_fire = FALSE
+			return FALSE
+		recalculate_atoms()
+		if(do_purge)
+			purging = FALSE
+			purge_tick++
+			if(purge_tick >= purge_interval)
+				purging_atoms = list()
+				purging = TRUE
+				purge_tick = 0
+	while(currentrun_manual.len)
+		var/atom/movable/AM = currentrun_manual[currentrun_manual.len]
+		currentrun_manual.len--
+		if(istype(AM))
+			AM.gravity_tick += wait
+			if(AM.gravity_tick >= AM.gravity_speed)
+				AM.manual_gravity_process()
+				AM.gravity_tick = 0
+		else
+			error_no_atom++
+		if(purging && do_purge)
```

Comment:
those are boolean variables, purging is for if it gets mc_tick_check'd during the purging and do_purge is whether it's doing it, i'm debating whether to leave it in because this is more of a "just incase anything got left in an error state this will get rid of them"